--- origsrc/libiptcdata-1.0.2/libiptcdata/Makefile.am	2005-08-28 00:18:55.000000000 -0500
+++ src/libiptcdata-1.0.2/libiptcdata/Makefile.am	2007-07-17 22:41:41.609375000 -0500
@@ -5,7 +5,8 @@
 
 lib_LTLIBRARIES = libiptcdata.la
 
-libiptcdata_la_LDFLAGS = -version-info @LIBIPTCDATA_VERSION_INFO@
+libiptcdata_la_LDFLAGS = -version-info @LIBIPTCDATA_VERSION_INFO@ -no-undefined
+libiptcdata_la_LIBADD = $(LTLIBINTL)
 libiptcdata_la_SOURCES =		\
 	iptc-data.c		\
 	iptc-dataset.c		\
--- origsrc/libiptcdata-1.0.2/python/Makefile.am	2007-05-15 01:12:59.000000000 -0500
+++ src/libiptcdata-1.0.2/python/Makefile.am	2007-07-17 20:55:43.703125000 -0500
@@ -3,6 +3,6 @@
 pyexec_LTLIBRARIES = iptcdata.la
 
 iptcdata_la_SOURCES = pyiptcdatamod.c pyiptcdata.c pyiptcdataset.c pyiptcdata.h
-iptcdata_la_LIBADD = ../libiptcdata/libiptcdata.la
-iptcdata_la_LDFLAGS = -module -avoid-version
+iptcdata_la_LIBADD = ../libiptcdata/libiptcdata.la -lpython$(PYTHON_VERSION)
+iptcdata_la_LDFLAGS = -module -avoid-version -no-undefined -L$(libdir)/python$(PYTHON_VERSION)/config 
 iptcdata_la_CPPFLAGS = $(PYTHON_CPPFLAGS) -I$(top_srcdir) -fno-strict-aliasing -DNDEBUG -g -O2 -Wall -Wstrict-prototypes
